### PR TITLE
[FIX] search_engine_image_thumbnail: fix the domain type for image fields filter

### PR DESCRIPTION
**Pull request by kobros-tech**
The domain attribute is provided in the backend in the field definition.

Providing the domain in the form view caused a critical frontend error prohibiting me from choosing the value.

### CODE REVIEW

**Review by lmignon — CHANGES_REQUESTED** [review]
Hi @kobros-tech,

Thank you for the proposal but IMO your change is not right. The domain on the `field_id` field is there to ensure that you can only select fields for the selected model on the UI form. I you remove the domain, you open the door to the selection of any fields of any models.

**Comment by kobros-tech** [comment]
> Hi @kobros-tech,
> 
> Thank you for the proposal but IMO your change is not right. The domain on the `field_id` field is there to ensure that you can only select fields for the selected model on the UI form. I you remove the domain, you open the door to the selection of any fields of any models.

@lmignon 
With the domain there is a critical error due to the domain itself, the domain is not correct it is just a binary field value.

I think the idea of filter is not applied correctly, but we can work on it to make a correct domain to filter out the correct values to choose from.

If you are not sure I can debug and show you the error again.

Thanks

**Comment by lmignon** [comment]
A binary field for a domain is the right way to provide a computed domain
to a field into a UI form. The domain itself could be wrong but this code
is right.

**Comment by kobros-tech** [comment]
@lmignon 

Thanks @lmignon for you patience.
Although I knew that the JSON binary field is including the correct domain, but I couldn't find any similar code in odoo to convert it into a string to be functional. 

I tried the same domain value included in the binary field by writing hard code and it was correct.

I do not argue, but honestly the domain was correct and your code to collect the domain was correct.

The only issue was the type of it. and I could solve the issue by creating a computed field many2many which includes all the "names" records the should be a condition in the domain.

This method succeeded without errors and I saw all behaviours in original odoo repository are doing the same as mine.

I am interested in making the module successful for my business and for the community so please understand my point and consider me as a junior developer but I am only interested in improving things whether who make it me or you or any other developer.

Here are the snapshots, and if you allow me I can improve the commit so we add my modification on the field that I tested.


Screenshot from 2024-10-07 17-46-09
Screenshot from 2024-10-07 17-45-44
Screenshot from 2024-10-07 15-25-49

Screenshot from 2024-10-07 13-58-16
Screenshot from 2024-10-07 13-57-40

File: search_engine_image_thumbnail/models/se_image_field_thumbnail_size.py [hunk 1]
```diff
@@ -93,26 +92,22 @@ def _is_model_valid_for_thumbnail(self, model_name):
...
             names = []
             for field in domain_fields:
                 if self._is_field_valid_for_thumbnail(field):
                     names.append(field.name)
-            record.field_id_domain = json.dumps(
-                [("name", "in", names), ("model_id", "=", record.model_id.id)]
-                if names
-                else FALSE_DOMAIN
-            )
-
```
Comment (lmignon) [inline search_engine_image_thumbnail/models/se_image_field_thumbnail_size.py:115]: @kobros-tech The error is here.... if you assign directly the domain to the field_id_domain without using the call to `json.dumps` this should work. This pattern is very common in many mocules in the community and it works like a charm. It's also used into Odoo.

```python
record.field_id_domain = (
    [("name", "in", names), ("model_id", "=", record.model_id.id)]
    if names
    else FALSE_DOMAIN
)

**Review by lmignon — CHANGES_REQUESTED** [review]

File: search_engine_image_thumbnail/models/se_image_field_thumbnail_size.py (hunk 1 above)
Comment (kobros-tech) [reply search_engine_image_thumbnail/models/se_image_field_thumbnail_size.py:115]: Yeah, it works now after removing dumps function.

Thanks @lmignon very much, I will keep both approaches but I will archive mine which is pulled here.

See you on the new pull request.

[https://github.com/OCA/search-engine/pull/200](https://github.com/OCA/search-engine/pull/200)

**Pull request closed without merging**